### ISSUE TITLE
Geany-CRITICAL **: sci_get_contents_range: assertion 'start < end' failed Segmentation fault

### ISSUE THREAD

**Issue by petermolnar** [geany/geany#1016]
I'm editing a file, which contains the following line:

``` php
        // match cite
        preg_match_all( '/<blockquote>(?:.*?)(?:&#8211;|–|\\-|<br />\w-)(.*?)(:?<\/p>)(?:.*)?<\/blockquote>/s', $r, $matches );
```

When trying to edit the text inside the regex, geany craches with the message in th title.

Environment: elementaryOS Freya, based on Ubuntu 14.04, Geany-dev ppa, installed versions:

```
geany
Version: 1.27-1~trusty
geany-common
Version: 1.27-1~trusty
geany-plugin-addons
Version: 1.27-1~trusty
geany-plugin-autoclose
Version: 1.27-1~trusty
geany-plugin-automark
Version: 1.27-1~trusty
geany-plugin-codenav
Version: 1.27-1~trusty
geany-plugin-commander
Version: 1.27-1~trusty
geany-plugin-ctags
Version: 1.27-1~trusty
geany-plugin-debugger
Version: 1.27-1~trusty
geany-plugin-defineformat
Version: 1.27-1~trusty
geany-plugin-devhelp
Version: 1.27-1~trusty
geany-plugin-doc
Version: 1.27-1~trusty
geany-plugin-extrasel
Version: 1.27-1~trusty
geany-plugin-gendoc
Version: 1.27-1~trusty
geany-plugin-geniuspaste
Version: 1.27-1~trusty
geany-plugin-git-changebar
Version: 1.27-1~trusty
geany-plugin-insertnum
Version: 1.27-1~trusty
geany-plugin-latex
Version: 1.27-1~trusty
geany-plugin-lipsum
Version: 1.27-1~trusty
geany-plugin-lua
Version: 1.27-1~trusty
geany-plugin-macro
Version: 1.27-1~trusty
geany-plugin-markdown
Version: 1.27-1~trusty
geany-plugin-miniscript
Version: 1.27-1~trusty
geany-plugin-multiterm
Version: 1.27-1~trusty
geany-plugin-numberedbookmarks
Version: 1.27-1~trusty
geany-plugin-overview
Version: 1.27-1~trusty
geany-plugin-pairtaghighlighter
Version: 1.27-1~trusty
geany-plugin-pg
Version: 1.27-1~trusty
geany-plugin-pohelper
Version: 1.27-1~trusty
geany-plugin-prettyprinter
Version: 1.27-1~trusty
geany-plugin-prj
Version: 1.27-1~trusty
geany-plugin-projectorganizer
Version: 1.27-1~trusty
geany-plugin-py
Version: 1.27-1~trusty
geany-plugin-scope
Version: 1.27-1~trusty
geany-plugin-sendmail
Version: 1.27-1~trusty
geany-plugin-shiftcolumn
Version: 1.27-1~trusty
geany-plugin-spellcheck
Version: 1.27-1~trusty
geany-plugin-tableconvert
Version: 1.27-1~trusty
geany-plugin-treebrowser
Version: 1.27-1~trusty
geany-plugin-updatechecker
Version: 1.27-1~trusty
geany-plugin-vc
Version: 1.27-1~trusty
geany-plugin-webhelper
Version: 1.27-1~trusty
geany-plugin-xmlsnippets
Version: 1.27-1~trusty
geany-plugins
Version: 1.27-1~trusty
geany-plugins-common
Version: 1.27-1~trusty
```

What else can I provide?

**Comment by elextr** [comment]
Can you try running Geany under GDB and when it crashes do a bt and post the results (note, make sure you get the whole trace, gdb has a bad habit of only showing part of it and requiring you to continue it)

**Comment by elextr** [comment]
Oh, and what language is that you are editing?  Not clear from just one line :)

**Comment by petermolnar** [comment]
The language is PHP; sorry, I was expecting it to show up, since I added that to the markdown fence.

This is the complete gdb output:
```
gdb geany 
GNU gdb (Ubuntu 7.7.1-0ubuntu5~14.04.2) 7.7.1
Copyright (C) 2014 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.  Type "show copying"
and "show warranty" for details.
This GDB was configured as "x86_64-linux-gnu".
Type "show configuration" for configuration details.
For bug reporting instructions, please see:
<http://www.gnu.org/software/gdb/bugs/>.
Find the GDB manual and other documentation resources online at:
<http://www.gnu.org/software/gdb/documentation/>.
For help, type "help".
Type "apropos word" to search for commands related to "word"...
Reading symbols from geany...(no debugging symbols found)...done.
(gdb) run
Starting program: /usr/bin/geany 
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib/x86_64-linux-gnu/libthread_db.so.1".
[New Thread 0x7fffec3da700 (LWP 25469)]
[New Thread 0x7fffebbd9700 (LWP 25470)]
[New Thread 0x7fffeb3d8700 (LWP 25471)]
[New Thread 0x7fffeabd7700 (LWP 25472)]
[Thread 0x7fffeabd7700 (LWP 25472) exited]
(geany:25465): Geany-CRITICAL **: sci_get_contents_range: assertion 'start < end' failed

Program received signal SIGSEGV, Segmentation fault.
__strcmp_sse2_unaligned () at ../sysdeps/x86_64/multiarch/strcmp-sse2-unaligned.S:30
30	../sysdeps/x86_64/multiarch/strcmp-sse2-unaligned.S: No such file or directory.
(gdb) bt
#0  __strcmp_sse2_unaligned () at ../sysdeps/x86_64/multiarch/strcmp-sse2-unaligned.S:30
#1  0x00007fffd1667545 in ?? () from /usr/lib/x86_64-linux-gnu/geany/pairtaghighlighter.so
#2  0x00007ffff79a65fa in ?? () from /usr/lib/x86_64-linux-gnu/libgeany.so.0
#3  0x00007ffff5b371c8 in g_closure_invoke () from /usr/lib/x86_64-linux-gnu/libgobject-2.0.so.0
#4  0x00007ffff5b483ed in ?? () from /usr/lib/x86_64-linux-gnu/libgobject-2.0.so.0
#5  0x00007ffff5b4fd88 in g_signal_emit_valist () from /usr/lib/x86_64-linux-gnu/libgobject-2.0.so.0
#6  0x00007ffff5b508a2 in g_signal_emit_by_name () from /usr/lib/x86_64-linux-gnu/libgobject-2.0.so.0
#7  0x00007ffff7998d6f in ?? () from /usr/lib/x86_64-linux-gnu/libgeany.so.0
#8  0x00007ffff5b371c8 in g_closure_invoke () from /usr/lib/x86_64-linux-gnu/libgobject-2.0.so.0
#9  0x00007ffff5b483ed in ?? () from /usr/lib/x86_64-linux-gnu/libgobject-2.0.so.0
#10 0x00007ffff5b500b8 in g_signal_emit_valist () from /usr/lib/x86_64-linux-gnu/libgobject-2.0.so.0
#11 0x00007ffff5b50372 in g_signal_emit () from /usr/lib/x86_64-linux-gnu/libgobject-2.0.so.0
#12 0x00007ffff79f317f in ?? () from /usr/lib/x86_64-linux-gnu/libgeany.so.0
#13 0x00007ffff7a15e32 in ?? () from /usr/lib/x86_64-linux-gnu/libgeany.so.0
#14 0x00007ffff7a1d43a in ?? () from /usr/lib/x86_64-linux-gnu/libgeany.so.0
#15 0x00007ffff79f3655 in ?? () from /usr/lib/x86_64-linux-gnu/libgeany.so.0
#16 0x00007ffff565fab5 in g_main_context_dispatch () from /lib/x86_64-linux-gnu/libglib-2.0.so.0
#17 0x00007ffff565fe18 in ?? () from /lib/x86_64-linux-gnu/libglib-2.0.so.0
#18 0x00007ffff56600da in g_main_loop_run () from /lib/x86_64-linux-gnu/libglib-2.0.so.0
#19 0x00007ffff6e51447 in gtk_main () from /usr/lib/x86_64-linux-gnu/libgtk-x11-2.0.so.0
#20 0x00007ffff79b6dd1 in main_lib () from /usr/lib/x86_64-linux-gnu/libgeany.so.0
#21 0x00007ffff7384ec5 in __libc_start_main (main=0x4005c0, argc=1, argv=0x7fffffffdf58, init=<optimized out>, fini=<optimized out>, 
    rtld_fini=<optimized out>, stack_end=0x7fffffffdf48) at libc-start.c:287
#22 0x00000000004005ee in ?? ()
```

There were loads of pixbuf error lines, but I think that is unrelated, because it's been like this for a few months (note: username replace with USERNAME):
```
** (geany:25465): WARNING **: Invalid borders specified for theme pixmap:
        /home/USERNAME/.themes/minimum-black-suite/gtk-2.0/Scrollbars/slider-vert.png,
borders don't fit within the image

** (geany:25465): WARNING **: Pixbuf theme: Cannot load pixmap file /home/USERNAME/.themes/minimum-black-suite/gtk-2.0/Tabs/gap-top.png: Failed to load image '/home/USERNAME/.themes/minimum-black-suite/gtk-2.0/Tabs/gap-top.png': Fatal error in PNG image file: Not a PNG file

** (geany:25465): WARNING **: Invalid borders specified for theme pixmap:
        /home/USERNAME/.themes/minimum-black-suite/gtk-2.0/Tabs/gap-top.png,
borders don't fit within the image


(geany:25465): GdkPixbuf-CRITICAL **: gdk_pixbuf_get_n_channels: assertion 'GDK_IS_PIXBUF (pixbuf)' failed

(geany:25465): GdkPixbuf-CRITICAL **: gdk_pixbuf_get_pixels: assertion 'GDK_IS_PIXBUF (pixbuf)' failed

(geany:25465): GdkPixbuf-CRITICAL **: gdk_pixbuf_get_rowstride: assertion 'GDK_IS_PIXBUF (pixbuf)' failed
```

**Comment by elextr** [comment]
Ok, the problem looks like its in the pairtaghighlighter plugin.  If the crash goes away if the plugin is disabled then please report on the [geany-plugins issues](https://github.com/geany/geany-plugins/issues)

**Comment by eht16** [comment]
This might be related to https://sourceforge.net/p/geany-plugins/bugs/130/ and maybe #989.

**Comment by fabriceh** [comment]
I confirm problem with pairtaghighlighter plugin.
I can reproduce with this small HTML file when deleting first **ul** (just letters, not < > )
```
<!DOCTYPE html>
 <html>
 <body>
 <ul >
 </ul>
 </body>
 </html>
```
append only if pairtaghighlighter plugin is active

**Comment by eht16** [comment]
What version of Geany-Plugins are you using? This bug should have been fixed in 1.28.
See https://github.com/geany/geany-plugins/issues/442 and  https://github.com/geany/geany-plugins/issues/446.

**Comment by LarsGit223** [comment]
I could not re-produce this with geany 1.33 and it seems that it was fixed in https://github.com/geany/geany-plugins/pull/447. So it can be closed I guess.
